dask/array/tests/test_reductions.py

```diff
@@ -809,3 +809,37 @@ def test_nan_func_does_not_warn(func):
     with pytest.warns(None) as rec:
         getattr(da, func)(d).compute()
     assert not rec  # did not warn
+
+
+@pytest.mark.parametrize("chunks", [((3, 3), (2, 1, 2, 1)), ((2, 1, 2, 1), (3, 3))])
+def test_chunk_structure_dependence(chunks):
+    # Reducing an array should not depend on its chunk-structure!!!
+    # See Issue #8541: https://github.com/dask/dask/issues/8541
+
+    def chunk_func(x_chunks, axis, keepdims):
+        return da.core._concatenate2(x_chunks, axes=axis)
+
+    a, b = 6, 6
+    array = np.arange(a * b).reshape(a, b)
+    x = da.from_array(array, chunks=chunks)
+    reduced_x = da.reduction(
+        x,
+        lambda x, axis, keepdims: x,
+        chunk_func,
+        keepdims=True,
+        concatenate=False,
```

Comment:
In a comment you gave a simpler test case that had `concatenate=True` - would you mind switching to that test case (if you think it's sufficient)?

Reply:
Sure.  No problem.